deeprankcore/dataset.py

```diff
@@ -581,24 +581,36 @@ def __init__( # pylint: disable=too-many-arguments, too-many-locals
                 Defaults to None.
             subset (Optional[List[str]], optional): List of keys from .HDF5 file to include. 
                 Defaults to None (meaning include all).
+                
+            dataset_train (class:`GraphDataset`, optional): If train is True, assign here the training set,
+                the following parameters target, task, node_features, edge_features and features_transform will be set
+                as the same value in dataset_train. 
+
+                Defaults to None.
+            
```

Comment:
I think I didnt explain well in my previous comment. 

Right now, it's still a bit confusing, especially because the `train` parameter is set later.

I think it makes sense to first set `train` to True or False and in the description of that argument state: if `True` then set these arguments and if `False` set the other arguments. Then in the description of the other arguments state that they will be ignored if the `train` setting is False/True.

Reply:
Thanks for the explanation:)
True that the order of the definition is quite confusing, 
I adjust the order of the parameters and the description in docstring a bit.
Now it should be more readable.